### PR TITLE
Move DbConnectionOptions.cs to Common

### PR DESCRIPTION
**Pull request by EgorBo**
Just like https://github.com/dotnet/corefx/pull/18500 this PR moves common code from three similar DbConnectionOptions.cs copies (SqlClient, Odbc, Common) to [src/Common/src/System/Data/Common](https://github.com/dotnet/corefx/tree/master/src/Common/src/System/Data/Common) folder.
I didn't remove any members but I had to rename some as these files use different code styles (internal fields) and Odbc and Common used to use `Hashtable `instead of `Dictionary<string, string>` for 'synonyms' (like SqlClient does). It's needed to make mono happy.

Probably would be better to just move DbConnectionOptions.cs to Common without splitting it to provider-specific pieces - they are not too big and the ILLINK will remove unused code anyway.

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @saurabh500, @corivera

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop OSX10.12 Release x64 Build and Test please

**Comment by danmosemsft** [comment]
cleanup= afte rfork.

**Comment by danmosemsft** [comment]
cleanup= afte rfork.

**Comment by EgorBo** [comment]
@danmosemsft sorry, but what does it mean?

**Comment by danmosemsft** [comment]
@EgorBo sorry, was on phone. I just meant, this merge can hold on for a couple days until 2.0 has branched off -- probably https://github.com/dotnet/corefx/pull/19891

**Comment by EgorBo** [comment]
@danmosemsft ah ok, makes sense.

**Comment by saurabh500** [comment]
@EgorBo I won't be able to get to this PR for a couple of days. 
I have seen similar PRs come in, in the past and merged. Thanks for consolidating the code. 
How will all this help Mono? Can you give me a synopsis of where this effort is leading to?

**Comment by saurabh500** [comment]
Looking at https://github.com/mono/mono/pull/4893/ and https://github.com/dotnet/corefx/pull/18500 it seems like you want to reuse the CoreFx Code for the Mono's System.Data.dll and this refactoring / code movement will allow much reuse of what exists. 
This seems important for the providers that are not there in CoreFx like ODBC and OleDb. Is this understanding correct?

**Comment by EgorBo** [comment]
@saurabh500 Exactly! System.Data in corefx is represented by 3 assemblies: Common, SqlClient and Odbc, but we try to use all these files inside a single assembly System.Data.dll (we can't split) - so I faced with several conflicts like this when each corefx System.Data assembly has *almost* identical internal types with some provider specific pieces (I already fixed most of those conflicts in my previous PRs - https://github.com/dotnet/corefx/pulls?q=is%3Apr+author%3AEgorBo+is%3Aclosed)

**Comment by karelz** [comment]
Just a heads up that System.Data is running hot to address all feedback on 2.0, so we may need to delay any response to PRs for couple of weeks. @EgorBo if that is not ok, please let us know ASAP. Thanks for understanding!

**Comment by EgorBo** [comment]
@karelz oh no rush, I temporarily moved these files to the mono repository. Thanks.

**Comment by karelz** [comment]
@saurabh500 @corivera do you have rough ETA when you will be able to look at the PR? (no pressure, just trying to understand the expectations ...)

**Comment by saurabh500** [comment]
@karelz we will take care of this pr today.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 namespace System.Data.SqlClient
 {
-    internal sealed class TdsParserStateObjectFactory
+    internal sealed partial class TdsParserStateObjectFactory
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs]: Why did you change this to partial?

**Comment by saurabh500** [comment]
@dotnet-bot Test this please

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs]: @EgorBo

**Comment by danmosemsft** [comment]
@dotnet-bot test this please (logs gone)

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs (hunk 1 above)
Comment (EgorBo) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs]: @saurabh500 @danmosemsft ah we don't need it actually (especially since we decided to use .Unix.cs version on Windows). Sorry I was on vacation last two weeks. Let me remove this change.

**Comment by saurabh500** [comment]
@karelz, i was waiting for a response to a question of mine. I will merge the PR once the CI is green.

**Comment by saurabh500** [comment]
He has taken care of the changes. We are good to go
